### PR TITLE
Add Count and ReferenceManyCount components

### PR DESCRIPTION
**Pull request by fzaninotto**
To count the number of records related to the current one via a one-to-many relationship, this PR introduces the `<ReferenceManyCount>` component.

Capture d’écran 2022-12-11 à 15 53 23

```jsx
import { 
    ChipField
    Datagrid,
    DateField,
    List,
    NumberField,
    ReferenceArrayField,
    ReferenceManyCount,
    SingleFieldList,
    TextField,
} from 'react-admin';

export const PostList = () => (
    <List>
        <Datagrid>
            <TextField source="id" />
            <TextField source="title" />
            <DateField source="published_at" sortByOrder="DESC" />
            <ReferenceManyCount
                label="Nb comments"
                reference="comments"
                target="post_id"
                link
            />
            <NumberField source="views" sortByOrder="DESC" />
            <ReferenceArrayField
                label="Tags"
                reference="tags"
                source="tags"
            >
                <SingleFieldList>
                    <ChipField source="name.en" size="small" />
                </SingleFieldList>
            </ReferenceArrayField>
        </Datagrid>
    </List>
)
```

To count the number of records in a resource, this PR introduces the `<Count>` component. 

Capture d’écran 2022-12-10 à 23 16 22

```jsx
import { Count, useStore } from 'react-admin';
import { MenuList, MenuItem, ListItemText } from '@mui/material';
import { isEqual } from 'lodash';

const TicketListAside = () => {
  const [statusFilter, setStatusFilter] = useStore("statusMenu", { status: 'open' });
  return (
    <MenuList>
        <MenuItem
            onClick={() => setStatusFilter({ status: 'open' })}
            selected={isEqual(statusFilter, { status: 'open' })}
        >
            <ListItemText>Open</ListItemText>
            <Count filter={{ status: 'open' }} />
        </MenuItem>
        <MenuItem
            onClick={() => setStatusFilter({ status: 'pending' })}
            selected={isEqual(statusFilter, { status: 'pending' })}
        >
            <ListItemText>Pending</ListItemText>
            <Count filter={{ status: 'pending' }} />
        </MenuItem>
        <MenuItem
            onClick={() => setStatusFilter({ status: 'closed' })}
            selected={isEqual(statusFilter, { status: 'closed' })}
        >
            <ListItemText>Closed</ListItemText>
            <Count filter={{ status: 'closed' }} />
        </MenuItem>
        <MenuItem
            onClick={() => setStatusFilter({})}
            selected={isEqual(statusFilter, {})}
        >
            <ListItemText>All</ListItemText>
            <Count filter={{}} />
        </MenuItem>
    </MenuList>
  );
};
```

### CODE REVIEW

**Review by djhi — APPROVED** [review]
:fire:

File: docs/navigation.html [hunk 1]
```diff
@@ -76,6 +76,7 @@
   <li {% if page.path == 'Pagination.md' %} class="active" {% endif %}><a class="nav-link" href="./Pagination.html"><code>&lt;Pagination&gt;</code></a></li>
   <li {% if page.path == 'SortButton.md' %} class="active" {% endif %}><a class="nav-link" href="./SortButton.html"><code>&lt;SortButton&gt;</code></a></li>
   <li {% if page.path == 'SelectColumnsButton.md' %} class="active" {% endif %}><a class="nav-link" href="./SelectColumnsButton.html"><code>&lt;SelectColumnsButton&gt;</code></a></li>
+  <li {% if page.path == 'Count.md' %} class="active" {% endif %}><a class="nav-link" href="./Count.html"><code>&lt;Count&gt;</code></a></li>
```
Comment (slax57) [inline docs/navigation.html:79]: Don't you need to add them to the Reference page too?

**Review by slax57 — DISMISSED** [review]
Otherwise I agree with djhi , it rocks! :fire:

File: docs/navigation.html (hunk 1 above)
Comment (fzaninotto) [reply docs/navigation.html:79]: You're right, I forgot! Fixed in the last commit.

**Pull request merged**